code/modules/flufftext/Hallucination.dm

```diff
@@ -626,24 +727,44 @@ GLOBAL_LIST_INIT(hallucination_list, list(
 		door_number = rand(0,4) //if 0 bolts all visible doors
 	var/count = 0
 	feedback_details += "Door amount: [door_number]"
+
 	for(var/obj/machinery/door/airlock/A in range(7, target))
 		if(count>door_number && door_number>0)
 			break
 		if(!A.density)
 			continue
 		count++
-		var/obj/effect/hallucination/fake_door_lock/lock = new(get_turf(A))
-		lock.target = target
-		lock.airlock = A
-		locks += lock
-		lock.lock()
-		sleep(rand(4,12))
-	sleep(100)
-	for(var/obj/effect/hallucination/fake_door_lock/lock in locks)
-		locks -= lock
-		lock.unlock()
-		sleep(rand(4,12))
-	qdel(src)
+		LAZYADD(airlocks_to_hit, A)
+
+	addtimer(VARSET_CALLBACK(src, unlocking, TRUE), 10 SECONDS)
+	START_PROCESSING(SSfastprocess, src)
+
+/datum/hallucination/bolts/process(delta_time)
+	next_action -= (delta_time * 10)
+	if (next_action > 0)
+		return
+
+	if (unlocking)
+		var/obj/effect/hallucination/fake_door_lock/next_unlock = popleft(locks)
+		if (next_unlock)
+			next_unlock.unlock()
+		else
+			qdel(src)
+			return
+	else
+		var/atom/next_airlock = pop(airlocks_to_hit)
```

Comment:
Why use 2 lists when you can use 1 and an index?

Reply:
One is for the hallucination object, and one is for the airlocks. I was just uncomfortable mixing the two types together.

Reply:
ah I see, that works fine.